lib/core/test/unit/Cardano/Wallet/Network/LightSpec.hs

```diff
@@ -250,26 +259,30 @@ updateFollower :: State s a -> MockMonad s a
 updateFollower action = Free (UpdateFollower action Pure)
 
 instance MonadDelay (MockMonad s) where
-    threadDelay _ = wait
+    threadDelay _ = wait <* void tick
 
 -- | Evaluate a 'MockMonad' action on a given 'ChainHistory'.
 evalMockMonad :: MockMonad s a -> ChainHistory -> s -> s
 evalMockMonad action0 (ChainHistory chain0 deltas0) s0
-    = go action0 s0 chain0 deltas0
+    = go (500 :: Integer) False action0 s0 chain0 deltas0
   where
-    go (Pure _) s _ _ = s -- monad has finished
-    go (Free action) s chain deltas = case action of
-        Wait k -> case deltas of
-            [] -> s -- chain will not change anymore
-            _ -> go (k ()) s chain deltas
-        Tick k -> case deltas of
-            [] -> go (k Idle) s chain deltas
-            ((d,chain2):ds) -> go (k d) s chain2 ds
-        GetChain k ->
-            go (k chain) s chain deltas
-        UpdateFollower act k ->
-            let (a, s2) = runState act s
-            in  go (k a) s2 chain deltas
+    go depth _lastWait _action _s _chain _deltas | depth < 1 =
+        error $ "Recursion depth exceeded (" <> show depth <> ")"
+    go _depth _waitCounter (Pure _) s _chain _deltas = s -- monad has finished
+    go (pred -> depth) lastWait (Free action) s chain deltas =
```

Comment:
```suggestion
    go d lastWait (Free action) s chain deltas = let depth = pred d in
```
Maybe?

I do find view patterns difficult to read. 🤔🤪

I would be much more comfortable with `go (Succ depth) …` where `Succ` is a pattern synonym.